common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -509,9 +552,146 @@ bool base_wlan_hal_dwpal::attach_ctrl_interface(int vap_id)
 
 bool base_wlan_hal_dwpal::process_nl_events()
 {
-    LOG(DEBUG) << __func__ << "isn't implemented by derived and shouldn't be called";
-    return false;
+    // process all NL events
+    if (!m_dwpal_nl_ctx) {
+        LOG(ERROR) << "Invalid WPA Control socket used for nl events (m_dwpal_ctx == nullptr)";
+        return false;
+    }
+
+    // Passing a lambda with capture is not supported for standard C function
+    // pointers. As a workaround, we create a static (but thread local) wrapper
+    // function that calls the capturing lambda function.
+    static __thread std::function<DWPAL_Ret(struct nl_msg * msg)> nl_handler_cb_wrapper;
+    nl_handler_cb_wrapper = [&](struct nl_msg *msg) -> DWPAL_Ret {
+        if (!process_dwpal_nl_event(msg)) {
+            LOG(ERROR) << "User's netlink handler function failed!";
+            return DWPAL_FAILURE;
+        }
+        return DWPAL_SUCCESS;
+    };
+    auto nl_handler_cb = [](struct nl_msg *msg) -> DWPAL_Ret { return nl_handler_cb_wrapper(msg); };
+
+    if (m_fd_nl_events > 0) {
```

Comment:
Shouldn't we print an error log if `m_fd_nl_events < 0` ?

Reply:
no, log file will explode (you enter here after_select)

Reply:
On the other hand  upper layer  (monitor_thread) has to check fd>0 before triggering process_nl_events().
So i'll add error print and return false .